ConfCore/Track.swift

```diff
@@ -43,22 +43,32 @@ public class Track: Object {
         return "name"
     }
 
-    public static func make(identifier: String,
-                            name: String,
-                            darkColor: String,
-                            lightBackgroundColor: String,
-                            lightColor: String,
-                            titleColor: String) -> Track {
-        let track = Track()
-
-        track.identifier = identifier
-        track.name = name
-        track.darkColor = darkColor
-        track.lightBackgroundColor = lightBackgroundColor
-        track.lightColor = lightColor
-        track.titleColor = titleColor
-
-        return track
+    // MARK: - Decodable
+
+    private enum CodingKeys: String, CodingKey {
+        case name, color, darkColor, titleColor, lightBGColor, ordinal
+        case identifier = "id"
     }
 
+    public convenience required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let identifier = try container.decode(Int.self, forKey: .identifier)
+        let name = try container.decode(String.self, forKey: .name)
```

Comment:
On another thing I was doing, I got tired of manually providing type information, so I wrote this:
```
extension KeyedDecodingContainer {

    func decode<T: Decodable>(forKey key: KeyedDecodingContainer.Key) throws -> T {
        return try decode(T.self, forKey: key)
    }

    func decodeIfPresent<T: Decodable>(forKey key: KeyedDecodingContainer.Key) throws -> T? {
        return try decodeIfPresent(T.self, forKey: key)
    }
}
```

Which then allows you to do `name = try container.decode(forKey: .name)`.

Also, not creating an intermediate variable (for simple decoding) would make the code more compact.